### PR TITLE
v4l2camera: add new package

### PR DESCRIPTION
**Pull request by mpromonet**
Maintainer: @mpromonet
Compile tested: (sunxi, nanopi neo air, OpenWrt latest)
Run tested: (sunxi, nanopi neo air, OpenWrt latest, enable service, start it and connect to http server, connect with vlc to rtsp server)

Description:
Add v4l2camera that provide a simple web server to RTSP server feeded with V4L2 capture device

### CODE REVIEW

File: multimedia/v4l2camera/files/v4l2camera.init [hunk 1]
```diff
@@ -0,0 +1,57 @@
+#!/bin/sh /etc/rc.common
+# Copyright (C) 2017 OpenWrt.org
```
Comment (1715173329) [inline multimedia/v4l2camera/files/v4l2camera.init:2]: please asisgn the copyright to yourself or just remove it

File: multimedia/v4l2camera/patches/001_live555_sprintf_format [hunk 2]
```diff
@@ -0,0 +1,25 @@
+From: Michel Promonet <michel.promonet@free.fr>
```
Comment (1715173329) [inline multimedia/v4l2camera/patches/001_live555_sprintf_format:1]: this patch file lacks a proper suffix: 001_live555_sprintf_format<strong>.patch</strong>

File: multimedia/v4l2camera/Makefile [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	HASH:=$(LIVE555_HASH)
+endef
+
+define Download/ui
+	URL:=https://registry.npmjs.org/v4l2camera-ui/-/
+	FILE:=$(UI_FILE)
+	HASH:=$(UI_HASH)
+endef
+
+define Package/v4l2camera/conffiles
```
Comment (1715173329) [inline multimedia/v4l2camera/Makefile:56]: please put this above `CMAKE_OPTIONS`

File: multimedia/v4l2camera/Makefile [hunk 4]
```diff
@@ -0,0 +1,87 @@
+# This software is in the public domain, furnished "as is", without technical
+# support, and with no warranty, express or implied, as to its usefulness for
+# any purpose.
+
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=v4l2camera
+PKG_VERSION:=0.1.8
+PKG_RELEASE:=2
```
Comment (1715173329) [inline multimedia/v4l2camera/Makefile:9]:
```suggestion
PKG_RELEASE:=1
```

this is a new package, so set it to `1`

**Review by 1715173329 — CHANGES_REQUESTED** [review]

File: multimedia/v4l2camera/files/v4l2camera.init (hunk 1 above)
Comment (mpromonet) [reply multimedia/v4l2camera/files/v4l2camera.init]: ok just removed

File: multimedia/v4l2camera/Makefile (hunk 3 above)
Comment (mpromonet) [reply multimedia/v4l2camera/Makefile]: moved

File: multimedia/v4l2camera/patches/001_live555_sprintf_format (hunk 2 above)
Comment (mpromonet) [reply multimedia/v4l2camera/patches/001_live555_sprintf_format]: renamed

File: multimedia/v4l2camera/Makefile [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+RSTRIP:=:
+
+define Package/v4l2camera
+	SECTION:=multimedia
+	CATEGORY:=Multimedia
+	TITLE:=v4l2camera
+	DEPENDS:=+libstdcpp 
+	URL:=https://github.com/mpromonet/v4l2camera
+endef
```
Comment (1715173329) [inline multimedia/v4l2camera/Makefile:38]:
```suggestion
define Package/v4l2camera
  SECTION:=multimedia
  CATEGORY:=Multimedia
  TITLE:=v4l2camera
  DEPENDS:=+libstdcpp 
  URL:=https://github.com/mpromonet/v4l2camera
endef
```

File: multimedia/v4l2camera/Makefile [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/${PKG_NAME} $(1)/usr/bin/
+
+	$(INSTALL_DIR) $(1)/usr/share/${PKG_NAME}
+	$(CP) $(PKG_INSTALL_DIR)/usr/share/${PKG_NAME} $(1)/usr/share/
+
+	$(INSTALL_DIR) $(1)/etc/init.d
+	$(INSTALL_BIN) files/${PKG_NAME}.init $(1)/etc/init.d/${PKG_NAME}
+	$(INSTALL_DIR) $(1)/etc/config
+	$(INSTALL_CONF) files/${PKG_NAME}.config $(1)/etc/config/${PKG_NAME}
```
Comment (1715173329) [inline multimedia/v4l2camera/Makefile:85]:
```suggestion
	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/v4l2camera $(1)/usr/bin/

	$(INSTALL_DIR) $(1)/usr/share/v4l2camera
	$(CP) $(PKG_INSTALL_DIR)/usr/share/v4l2camera $(1)/usr/share/

	$(INSTALL_DIR) $(1)/etc/init.d
	$(INSTALL_BIN) files/v4l2camera.init $(1)/etc/init.d/v4l2camera
	$(INSTALL_DIR) $(1)/etc/config
	$(INSTALL_CONF) files/v4l2camera.config $(1)/etc/config/v4l2camera
```

**Review by 1715173329 — APPROVED** [review]
otherwise LGTM

**Comment by mpromonet** [comment]
Hi,

I made an update to allow to configure more parameters in config file.

Best Regards,
Michel.

**Pull request merged**